PasswordValidator_UnitTests/AdvancedValidator_UnitTests.cs

```diff
@@ -6,14 +6,14 @@ namespace PasswordValidator_UnitTests
 {
     internal class AdvancedValidator_UnitTests
     {
-        AdvancedValidatorFactory _factory = new AdvancedValidatorFactory();
+        IValidatorFactory _factory = new PasswordValidatorFactory();
```

Comment:
In AdvancedValidator_UnitTests - you should just create an instance of the concrete class "AdvancedValidator" - you should have no reference to the factory. These tests should NOT know the factory exists.